### PR TITLE
Added EntryPoint document & minor changes

### PR DESCRIPTION
**Pull request by Chris-Schnaufer**
## Description

Added EntryPoint documentation and some other small changes

## Types of changes

New documentation
Fixed typos and some clarification

## Checklist:

- [x] I have read the documentation on [CONTRIBUTING](https://github.com/AgPipeline/Organization-info) 
- [x] My updates require a change to the documentation 
- [x] I have updated the documentation accordingly
- [ ] I have added tests to cover my changes
- [ ] All new and existing tests passed

## Time frame for completing Pull Request review (from creation date):
- [ ] As soon as possible
- [x] In the next two weeks
- [ ] In the next month

## Focus of Pull Request:
All comments are welcome regardless of which items are checked off.
- [ ] All of the below (no need to check off the other boxes)
- [ ] There are no errors or inconsistencies
- [x] Everything appears clear and easy to understand
- [ ] Suggest improvements
- [ ] Identify areas where functionality and/or documentation is missing

## Hints to reviewers:

There's a ticket for changing this documentation when the AgPypeline library is updated: https://app.zenhub.com/workspaces/ua-ag-data-science-5a57a3198339f11ba1c85775/issues/agpipeline/issues-and-projects/166

### CODE REVIEW

File: transformers/entrypoint.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
+# Entry Point
+This is the name given to the portion of a transformer that provides the flow of control through a Transformer and can provide some core common functionality.
+
+Please read the [Transformers](https://agpipeline.github.io/transformers/transformers) overview documentation for some additional context.
+
+In this document we will be providing an conceptual overview of the Entry Point code structure and then uses the implementation of the UA makeflow solution as a practical example.
```
Comment (dlebauer) [inline transformers/entrypoint.md]:
```suggestion
In this document we will be providing an conceptual overview of the Entry Point code structure and then use the implementation of the UA makeflow solution as a practical example.
```

File: transformers/entrypoint.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
+# Entry Point
```
Comment (dlebauer) [inline transformers/entrypoint.md:1]:
```suggestion
# Entry Point

```

File: transformers/entrypoint.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+This is the name given to the portion of a transformer that provides the flow of control through a Transformer and can provide some core common functionality.
+
+Please read the [Transformers](https://agpipeline.github.io/transformers/transformers) overview documentation for some additional context.
+
+In this document we will be providing an conceptual overview of the Entry Point code structure and then uses the implementation of the UA makeflow solution as a practical example.
+
+## Overview
```
Comment (dlebauer) [inline transformers/entrypoint.md:9]:
```suggestion
## Overview

```

File: transformers/entrypoint.md [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+By providing a common flow of execution through all Transformers the task of implementing a complete Transformer is simplified.
+Developers can focus on [Environment](https://agpipeline.github.io/transformers/environment) and/or [Algorithm](https://agpipeline.github.io/transformers/algorithm) specifics without worrying how to hook them together.
+
+The Entry Point Type can also provide some common, core functionality that is to be available for all Transformers.
+This core functionality can be directly implemented in the Entry Point code; such as requiring at least one file to be specific a command line.
+
+The flow of control through Entry Point is not expected to be immutable.
+The flow of control can be altered by definitions, function declarations, and other means.
+
+## AgPipeline Implementation
```
Comment (dlebauer) [inline transformers/entrypoint.md:21]:
```suggestion
## AgPipeline Implementation

```

**Review by dlebauer — COMMENTED** [review]
Overall, the documentation is dense! 

Is the intended audience limited to the developers of the infrastructure itself as opposed to scientists creating transformers?

If this is aimed at scientists, it would also be helpful to define cs terms like "immutable" and "flow of control". 

Perhaps it could be easier to read if each section started with a clear definition of each term _and_ a link to the relevant definition (on wikipedia perhaps? or in the documentation of relevant software?). For example, does the term 'entry point' have the same meaning as the docker entrypoint? If so, then be clear, and link here: https://docs.docker.com/engine/reference/builder/#entrypoint or is to this more generic concept on wikipedia? https://en.wikipedia.org/wiki/Entry_point or this python specific concept on the same page? https://en.wikipedia.org/wiki/Entry_point#Python

**Comment by Chris-Schnaufer** [comment]
> Overall, the documentation is dense

@dlbauer I've added an issue for this: https://github.com/AgPipeline/issues-and-projects/issues/184

**Pull request merged**